docs/en/tutorials/dataset.md

```diff
@@ -0,0 +1,200 @@
+
+In this tutorial, we will step through their usage in MMEngine runner from outside (dataloader) to inside (dataset) and give some practical examples. After reading through this tutorial, you will be able to:
+
+- Master the configuration of dataloaders in MMEngine
+- Learn to use existing datasets (e.g. those from `torchvision`) from config files
+- Know about building and using your own dataset
+
+## Details on dataloader
+
+Dataloaders can be configured in MMEngine's `Runner` with 3 arguments:
+
+- `train_dataloader`: Used in `Runner.train()` to provide training data for models
+- `val_dataloader`: Used in `Runner.val()` or in `Runner.train()` at regular intervals for model evaluation
+- `test_dataloader`: Used in `Runner.test()` for the final test
+
+MMEngine has full support for PyTorch native `DataLoader` objects. Therefore, you can simply pass your valid, already built dataloaders to the runner, as shown in [getting started in 15 minutes](../get_started/15_minutes.md). Meanwhile, thanks to the [Registry Mechanism](../advanced_tutorials/registry.md) of MMEngine, those arguments also accepts `dict`s as inputs, as illustrated in the following example (referred to as example 1). The keys in the dictionary correspond to arguments in DataLoader's init function.
+
+```python
+runner = Runner(
+    train_dataloader=dict(
+        batch_size=32,
+        sampler=dict(
+            type='DefaultSampler',
+            shuffle=True),
+        dataset=torchvision.datasets.CIFAR10(...),
+        collate_fn=dict(type='default_collate')
+    )
+)
+```
+
+When passed to the runner in the form of a dict, the dataloader will be lazily built in the runner when actually needed.
+
+```{note}
+For more configurable arguments of the `DataLoader`, please refer to [PyTorch API documentation](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader)
+```
+
+```{note}
+If you are interested in the details of the building procedure, you may refer to [build_dataloader](mmengine.runner.Runner.build_dataloader)
+```
+
+You may find example 1 differs from that in [getting started in 15 minutes](../get_started/15_minutes.md) in some arguments. Indeed, due to some obscure conventions in MMEngine, you can't seamlessly switch it to a dict by simply replace `DataLoader` with `dict`. We will discuss the differences between our convention and PyTorch's in the following sections, in case you run into trouble when using config files.
+
+### sampler and shuffle
+
+One obvious difference is that we add a `sampler` argument to the dict. This is because we **require `sampler` be explicitly specified** when using dict as dataloader. Meanwhile, `shuflle` is also removed from `DataLoader` arguments, because it conflicts with `sampler` in PyTorch, as referred to [PyTorch DataLoader API docs](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader).
```

Comment:
```suggestion
One obvious difference is that we add a `sampler` argument to the dict. This is because we **require `sampler` to be explicitly specified** when using dict as dataloader. Meanwhile, `shuflle` is also removed from `DataLoader` arguments, because it conflicts with `sampler` in PyTorch, as referred to [PyTorch DataLoader API docs](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader).
```